airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -636,6 +636,14 @@ def list(self, fields) -> Iterable:
             yield record
 
 
+class DealToContactAssociationsStream(CRMObjectStream):
```

Comment:
@vladimir-remar it seems better from a UX standpoint to include the deal-to-contact association as a field on the `deals` stream. This way in the destination it will be normalized to a `deal_contacts` table. WDYT?

Reply:
Hi @sherifnada  thanks for the review, about the suggestion make sense for me, I will include the field as "associations", since "contacts" are one of them , for me it would be something like this.

```
class DealStream(CRMObjectStream):
    """Deals, API v3"""

    def __init__(self, associations: List[str] = None, **kwargs):
        super().__init__(entity="deal", associations=associations, **kwargs)
        self._stage_history = DealStageHistoryStream(**kwargs)

    def list(self, fields) -> Iterable:
        history_by_id = {}
        for record in self._stage_history.list(fields):
            if all(field in record for field in ("id", "dealstage")):
                history_by_id[record["id"]] = record["dealstage"]
        for record in super().list(fields):
            if record.get("id") and int(record["id"]) in history_by_id:
                record["dealstage"] = history_by_id[int(record["id"])]
            yield record

```

and I will call it in `client.py`
`"deal_to_contact_associations": DealStream(associations=["contacts"], **common_params),`

Reply:
sorry I was a little unclear: why wouldn't we include this by default in the `Deals` stream instead of a separate `deals_to_contacts_associations`? That would be my first inclination although there may be a reason why that's not favorable.

Reply:
@vladimir-remar that approach (including associations=['contacts']) is the one I'd recommend!